### PR TITLE
Oneof refactor 2

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

Factor out some basic dataclasses so they can be used across multiple Test Classes.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/arcaflow_plugin_sdk/test_schema.py [hunk 1]
```diff
@@ -531,26 +555,25 @@ def test_unserialize(self):
             s_type.unserialize([])
         # Mismatching key value
         with self.assertRaises(ConstraintException):
-            s_type.unserialize({
-                self.discriminator_default: "a", "b": "Hello world!"})
+            s_type.unserialize(
+                {discriminator_default: "a", "b": "Hello world!"}
+            )
```
Comment (webbnh) [inline src/arcaflow_plugin_sdk/test_schema.py:560]: "discriminator_default" reads kind of funny, here (and my attempt to describe why reads even funnier, so I've redacted it).  The point is, wouldn't it sound better in this usage if it were `default_discriminator`?

**Review by webbnh — APPROVED** [review]
Looks great!  I just have a couple of thoughts for your consideration.

File: src/arcaflow_plugin_sdk/test_schema.py [hunk 2]
```diff
@@ -508,8 +533,7 @@ def test_unserialize_error_discriminator_type(self):
             scope=self.scope_basic,
         )
         with self.assertRaises(ConstraintException):
-            s_type.unserialize({
-                self.discriminator_default: "1", 1: "Hello world!"})
+            s_type.unserialize({discriminator_default: "1", 1: "Hello world!"})
```
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:536]: moved it and added a comment and code to make it more explicit

File: src/arcaflow_plugin_sdk/test_schema.py (hunk 1 above)
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:560]: yeah it reads funny, but it's also shorter than `discriminator_field_name_default`

**Review by dbutenhof — APPROVED** [review]
Probably OK, if it quacks like a duck, but maybe I'm just confused about the context of some type references. (Which I *thought* I'd posted yesterday, although clearly I didn't...)

File: src/arcaflow_plugin_sdk/test_schema.py (hunk 1 above)
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py]: renamed

File: src/arcaflow_plugin_sdk/test_schema.py [hunk 3]
```diff
@@ -15,6 +15,52 @@
...
+
+
+@dataclasses.dataclass
+class StrBasic:
+    msg: str
+
+
+@dataclasses.dataclass
+class StrBasic2:
+    msg2: int
```
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:35]: oh yeah you're right; i forgot about the confusion between naming it for the discriminator type vs the attribute

[the remark this answers is not in the document]
Comment (jaredoconnell) [reply src/arcaflow_plugin_sdk/test_schema.py:35]: Is the class name referring to if it's a part of a OneOf string, or if it contains a string?

[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:35]: oh that was a mistake

File: src/arcaflow_plugin_sdk/test_schema.py [hunk 4]
```diff
@@ -1451,123 +1481,135 @@ class TestData:
...
             ),
-            "A": schema.ObjectType(
-                A, {"a": schema.PropertyType(schema.StringType())}
+            StrBasic.__name__: schema.ObjectType(
+                StrBasic, {"msg": schema.PropertyType(schema.StringType())}
             ),
-            "B": schema.ObjectType(
-                B, {"b": schema.PropertyType(schema.StringType())}
+            StrBasic2.__name__: schema.ObjectType(
+                StrBasic2, {"msg2": schema.PropertyType(schema.StringType())}
```
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:1505]: fixed it

**Review by jaredoconnell — APPROVED** [review]

File: src/arcaflow_plugin_sdk/test_schema.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py]: @jaredoconnell that's a fair question

File: src/arcaflow_plugin_sdk/test_schema.py [hunk 5]
```diff
@@ -550,28 +540,35 @@ def test_unserialize(self):
         with self.assertRaises(ConstraintException):
             s_type.unserialize({"b": 42})
 
-        # invalid type for 'data' argument
+        # Invalid type for discriminator value
+        with self.assertRaises(ConstraintException):
+            discriminator_value = "O"
+            s_type.unserialize(
+                {default_discriminator: discriminator_value,
+                 1: "Hello world!"})
```
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py:548]: code as documentation to make it clear which identifier has the incorrect type

[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/test_schema.py]: ok i changed it

**Review by webbnh — APPROVED** [review]
Works for me!

**Review by dbutenhof — APPROVED** [review]

**Review by jaredoconnell — APPROVED** [review]

**Pull request merged**